### PR TITLE
Ex5.2 piece of cake

### PR DESCRIPTION
**Pull request by racheldekel**

### CODE REVIEW

File: ex5.1_this_is_the_way.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import os
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: Add a newline after imports

File: ex5.1_this_is_the_way.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+INPUT_FOLDER_MESSAGE="insert the path to the folder:"
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: Start messages to the user with a capital letter

File: ex5.1_this_is_the_way.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# function that print the file in folder that start with "deep"
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: Capture
that's the standard format of python documentation  - use it

File: ex5.1_this_is_the_way.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+def print_deep_folder():
+    folder_path = input(INPUT_FOLDER_MESSAGE)
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: Why export it to a const if you use it once? its more lines of code and makes the code less readable with no benefit

File: ex5.1_this_is_the_way.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    directory_list = os.listdir(folder_path)
+    only_deep = [f for f in directory_list if f.startswith("deep")]
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: only_deep_what? I would name this variable deep_files

Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: f is not a good variable name - change it

Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: there are things other then files in a folder - remember that

File: ex5.1_this_is_the_way.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    print(only_deep)
+
+#print the program
+if __name__ == '__main__':
+    print_deep_folder()
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: Add a newline at eof

File: ex5.2_cup_of_join.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#funcation that add the seperator for each list
+def print_join(*lists, seperator='-'):
+    list_with= [cell for curr_list in lists for cell in curr_list + [seperator]][:-1]
+    if list_with ==[]:
+        return 'empty'
+    return list_with
+
+if __name__ == '__main__':
+    print(print_join([1,2],[8],[9,5,6],seperator='@'))
```
Comment (talbarfm) [inline ex5.2_cup_of_join.py]: spaces after ,

File: ex5.2_cup_of_join.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#funcation that add the seperator for each list
+def print_join(*lists, seperator='-'):
+    list_with= [cell for curr_list in lists for cell in curr_list + [seperator]][:-1]
```
Comment (talbarfm) [inline ex5.2_cup_of_join.py]: cell is a bad variable name unless we are studying biology ;)

File: ex5.2_cup_of_join.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    if list_with ==[]:
+        return 'empty'
```
Comment (talbarfm) [inline ex5.2_cup_of_join.py]: you should avoid returning different types from functions even though python allows it -  https://stackoverflow.com/questions/1839289/why-should-functions-always-return-the-same-type

File: ex5.2_cup_of_join.py (hunk 7 above)
Comment (talbarfm) [inline ex5.2_cup_of_join.py]: newline at eof

File: ex5.2_piece_of_cake.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+#function that calculate the price for all the integers
+def get_recipe_price(prices=dict, optionals=list(), **quantity):
+    sum_price = 0
+    for key in prices.keys():
+            if key not in optionals:
```
Comment (talbarfm) [inline ex5.2_piece_of_cake.py]: reduce this to a oneline if

File: ex5.2_piece_of_cake.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
+#function that calculate the price for all the integers
+def get_recipe_price(prices=dict, optionals=list(), **quantity):
+    sum_price = 0
+    for key in prices.keys():
```
Comment (talbarfm) [inline ex5.2_piece_of_cake.py]: you can iterate over keys and values of a dictionary at once 
image

File: ex5.2_piece_of_cake.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+            if key not in optionals:
+                sum_price += (quantity[key] / 100 * prices[key])
+    return sum_price
+
+
+if __name__ == '__main__':
+    print(get_recipe_price({'chocolate': 18, 'milk': 8}, chocolate=200, milk=100))
```
Comment (talbarfm) [inline ex5.2_piece_of_cake.py]: Add a newline at eof

File: ex5.1_this_is_the_way.py [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import os
+INPUT_FOLDER_MESSAGE="insert the path to the folder:"
+# function that print the file in folder that start with "deep"
+def print_deep_folder():
+    folder_path = input(INPUT_FOLDER_MESSAGE)
+    directory_list = os.listdir(folder_path)
+    only_deep = [f for f in directory_list if f.startswith("deep")]
+    print(only_deep)
+
+#print the program
```
Comment (talbarfm) [inline ex5.1_this_is_the_way.py]: you dont have to document this part

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Comment by racheldekel** [comment]
fix

**Review by lovodkin93 — APPROVED** [review]
@racheldekel  good job! you can now merge the pull request.

**Pull request merged**